### PR TITLE
fix export IconNode type

### LINKED ISSUE #1654: Export the `IconNode` type
### Package

- [X] lucide
- [ ] lucide-angular
- [ ] lucide-flutter
- [ ] lucide-preact
- [ ] lucide-react
- [ ] lucide-react-native
- [ ] lucide-solid
- [ ] lucide-svelte
- [ ] lucide-vue
- [ ] lucide-vue-next
- [ ] Figma plugin
- [ ] all JS packages
- [ ] site

### Description

Hi! Thanks a lot for this great project.

As a user of Lucide in a component library, I want to add the possibility for our users to pass Lucide icon objects to some of the widgets `icon` property. Unfortunately, the `IconNode` seems to only be used internally. Would it possible to export it for TS projects?

(I also have [another question](https://github.com/lucide-icons/lucide/discussions/1594) if you feel like it 😁)

### Use cases

Consuming Lucide objects from projects.

### Checklist

- [X] I have searched the existing issues to make sure this bug has not already been reported.

### PR DESCRIPTION
**Pull request by Sukomal07**
fixes #1654

### CODE REVIEW

File: packages/lucide/rollup.config.mjs [hunk 1]
```diff
@@ -5,7 +5,7 @@ import pkg from './package.json' assert { type: 'json' };
 
 const outputFileName = pkg.name;
 const outputDir = 'dist';
-const inputs = ['src/lucide.ts'];
+const inputs = ['src/lucide.ts', 'src/types.ts'];
```
Comment (ericfennis) [inline packages/lucide/rollup.config.mjs:8]: Please revert this.

**Review by ericfennis — CHANGES_REQUESTED** [review]
I've just tested this. Unfortunately this is not going to work, since these inputs are used to bundle. In the bottom of this file we bundle the types, and that can be only one file.


So what we should do is export the types in: `packages/lucide/src/lucide.ts`.
Put this line at the bottom and this is solved.
```ts
/*
 Types exports.
*/
export * from './types';
```

**Comment by Sukomal07** [comment]
@ericfennis done

**Review by ericfennis — APPROVED** [review]
Awesome work!

**Comment by Sukomal07** [comment]
> Awesome work!

Thanks for help

**Comment by olaf-k** [comment]
that was fast, thanks a lot guys!

**Pull request merged**